cluster-autoscaler/core/equivalence_groups.go

```diff
@@ -87,3 +87,22 @@ func groupPodsBySchedulingProperties(pods []*apiv1.Pod) map[equivalenceGroupId][
 
 	return podEquivalenceGroups
 }
+
+func podSpecSemanticallyEqual(p1 apiv1.PodSpec, p2 apiv1.PodSpec) bool {
```

Comment:
This seems generally useful, not just for this cache. Maybe consider moving it to utils?

Reply:
Sorry, I meant cluster-autoscaler/utils. core/utils tend to be very hard to use outside /core (and particularly in processors) because of cyclic import problems.